src/components/MyKiva/BadgeInProgress.vue

```diff
@@ -0,0 +1,264 @@
+import useBadgeModal, {
+	MOBILE_BREAKPOINT,
+	BADGE_IN_PROGRESS,
+} from '#src/composables/useBadgeModal';
+import { useRouter } from 'vue-router';
+import useIsMobile from '#src/composables/useIsMobile';
+import BadgeContainer from './BadgeContainer';
+
+const props = defineProps({
+	/**
+	 * {
+	 *   id: '',
+	 *   fields: {
+	 *     challengeName: '',
+	 *     shareFact: '',
+	 *     badgeImage: {
+	 *       fields: {
+	 *         file: {
+	 *           url: '',
+	 *         },
+	 *       },
+	 *     },
+	 *   },
+	 *   totalProgressToAchievement,
+	 *   tiers: [
+	 *     {
+	 *       target: 2,
+	 *       completedDate: null,
+	 *     },
+	 *   ],
+	 * }
+	 */
+	badge: {
+		type: Object,
+		default: () => ({}),
+	},
+	isLoading: {
+		type: Boolean,
+		default: false
+	},
+	loanIds: {
+		type: Array,
+		default: () => ([])
+	},
+	loanLimit: {
+		type: Number,
+		default: 1
+	},
+	showViewMoreCard: {
+		type: Boolean,
+		default: false
+	}
+});
+
+const isLoading = computed(() => props.isLoading ?? false);
+const loanIds = computed(() => props.loanIds ?? []);
+const loanLimit = computed(() => props.loanLimit ?? 1);
+const loanDisplayCount = ref(3);
+const loadMoreClicked = ref(false);
+const kvCarouselRef = ref(null);
+const $kvTrackEvent = inject('$kvTrackEvent');
+const carouselKey = ref('badge-in-progress-carousel');
+const startIndex = ref(0);
+const {
+	getBadgeShape,
+	getPrefilteredUrl,
+} = useBadgeModal(props.badge);
+const { isMobile } = useIsMobile(MOBILE_BREAKPOINT);
+
+const router = useRouter();
+const forceRerender = () => {
+	carouselKey.value += 1;
+	startIndex.value = 1;
+};
+
+watch(loanIds, async ids => {
+	if (!isLoading.value) {
+		const defaultInitialCount = 3;
+		const loansCount = ids.length;
+		loanDisplayCount.value = loansCount > defaultInitialCount ? defaultInitialCount : loansCount;
+		loadMoreClicked.value = loansCount === loanDisplayCount.value;
+		kvCarouselRef.value?.reInitVisible();
+	}
+}, { immediate: true });
+
+const augmentedLoanIds = computed(() => {
+	if (isLoading.value) {
+		return [...Array(6).fill(0)];
+	}
+	return [...loanIds.value].splice(0, loanDisplayCount.value);
+});
+
+const viewAllLoansSlideIndex = computed(() => {
+	const slideIndex = augmentedLoanIds?.value?.length ?? 0;
+	return `slide${slideIndex + 1}`;
+});
+
+const singleSlideWidth = computed(() => {
+	const viewportWidth = typeof window !== 'undefined' ? window.innerWidth : 1024;
+	// Handle tiny screens
+	if (viewportWidth < 414) {
+		return `${viewportWidth - 80}px`;
+	}
+	if (viewportWidth >= 414 && viewportWidth < 768) return '278px';
+	if (viewportWidth >= 768 && viewportWidth < 1024) return '328px';
+	return '328px';
+});
+
+const onInteractCarousel = interaction => {
+	$kvTrackEvent('carousel', 'click-carousel-horizontal-scroll', interaction);
+};
+
+const badgeName = computed(() => props.badge?.fields?.challengeName ?? '');
+const badgeImage = computed(() => props.badge?.fields?.badgeImage?.fields?.file?.url ?? '');
```

Comment:
This will need to change once we update to use all badge icons, this needs to be the badge of the current level. I'm creating utils to get the correct tier. Please add a `TODO` here to use the icon of the correct in-progress badge level.